gangplank/internal/ocp/return.go

```diff
@@ -41,66 +41,81 @@ func (r *Return) Run(ctx context.Context, ws *workSpec) error {
 	if r.Minio == nil {
 		return nil
 	}
-	baseBuildDir := filepath.Join(cosaSrvDir, "builds")
-	b, path, err := cosa.ReadBuild(baseBuildDir, "", cosa.BuilderArch())
-	if err != nil {
-		return err
-	}
-	if b == nil {
-		return nil
-	}
 	upload := make(map[string]string)
 
-	// Capture /srv/builds/builds.json
+	// Grab any files that were explicitly placed here for us to
+	// upload denoted by having a .minioupload suffix. These get
+	// placed in the bucket under the minioupload/ prefix.
+	path := filepath.Join(cosaSrvDir, "tmp")
+	tmpFiles, _ := ioutil.ReadDir(path)
+	for _, f := range tmpFiles {
+		if strings.HasSuffix(f.Name(), "minioupload") {
+			upKey := filepath.Join("minioupload", f.Name())
+			srcPath := filepath.Join(path, f.Name())
+			upload[upKey] = srcPath
+		}
+	}
+
+	// Read and process the uploads for the builds. If there are no
+	// builds, ignore.
```

Comment:
This conflicts with the stage declarative nature of Gangplank (https://github.com/coreos/coreos-assembler/blob/main/gangplank/internal/spec/stages.go#L31-L98) in that each stage has explicit dependencies and the returned data is driven either by request in the stage or as defined meta-data.

I'd strongly prefer that the stage definition be extended to include `ReturnFiles` or something akin to that is processed via https://github.com/coreos/coreos-assembler/blob/main/gangplank/internal/ocp/worker.go#L225-L280

Reply:
seems fair will try in a new PR